New reduced Web SDK documentation (#1636)
SoftwareAG/c8y-docs: content/web/overview.md

* communicate authenticated with our API,
* apply default or branded UI components to your custom application.

{{< c8y-admon-important >}}
 Web SDK for Angular JS is deprecated. Therefore, its documentation is no longer provided here.  As all {{< product-c8y-iot >}} REST APIs are backward compatible, Angular JS applications will still continue to work.

Web SDK for plugins is based on Angular JS and is also deprecated. For Angular-based development, we recommend you to implement native Angular modules. You can import plugins to Angular as described in [Extension points](/web/libraries/#extension-points).
{{< /c8y-admon-important >}}

> ### When should I use the Web SDK?

Reviewer: ```suggestion
### When can I use the Web SDK?
```

The Web SDK is designed to extend default application or build new IoT solutions. It provides many default components to compose such applications. However, if one of our default applications already fits your usecase, you should think about extending it via the less-code approach and install plugins to them. If you simply have very special needs or can't find the right plugin for you use-case, consider building a custom application with the help of the Web SDK. There you basically have three options:
 1. Extend existing applications
 2. Build a new application with the help of the Web SDK
 3. Build a application from scratch

> Option 3. is the hardest option as you need to build nearly everything new. But you are free to choose your frameworks and tools. To support you on either of the options, we have developed and published certain packages to npm. Those packages help you to connect to our data, provides you with default components or help you test and deploy the app. Depending on your needs, you can decide which package you want to include. However we always encourage everyone to start with our boilerplates and the full Web SDK. You can find a description on how to use them in the [Getting started](/web/gettingstarted/) guide.

Reviewer: ```suggestion
The third option is the hardest option as you need to build nearly everything new. However, you are free to choose your frameworks and tools. To support you on either of the options, we have developed and published certain packages to npm. Those packages help you to connect to our data, provides you with default components or help you test and deploy the app. Depending on your needs, you can decide which package you want to include. However we always encourage everyone to start with our boilerplates and the full Web SDK. You can find a description on how to use them in the [Getting started](/web/gettingstarted/) guide.
```

The goal of these splittings is to provide the right package for every use case, e.g. if you want to build a small application with React you could use the `@c8y/client` to do the API interaction. If you need a brandable feature rich application which is close to our Cockpit or Device Mangement application you could use `@c8y/ngx-components` together with `@c8y/stlyes`.

Following is a list which explains the use cases of each package.

* [@c8y/client](https://www.npmjs.com/package/@c8y/client): Use this client to access our API. The client is isomorphic, that means it could be used in node.js and in the browser.
* [@c8y/ngx-components](https://www.npmjs.com/package/@c8y/ngx-components): A components collection and data access layer for Angular applications. This package can be used to build Angular applications.
* [@c8y/style](https://www.npmjs.com/package/@c8y/style): The styles for the look & feel of an application. Extend this package to apply a custom branding to your application.
* @c8y/&lt;&lt;apps&gt;&gt;: The different applications. E.g. `@c8y/cockpit`, `@c8y/devicemanagement` and `@c8y/administraiton`.

> * [@c8y/websdk](https://www.npmjs.com/package/@c8y/): The scaffolding package that you can use to add the Cumulocity applicaiton to your Angular application.

Reviewer 2: ```suggestion
* [@c8y/websdk](https://www.npmjs.com/package/@c8y/websdk): The scaffolding package that you can use to add the Cumulocity applicaiton to your Angular application.
```

 - Fix version (e.g. x.x.**7**): Cotains only fixes that should not break anything.

{{< c8y-admon-info >}}
As npm and semantic versioning only support three parts in the version number, the WebSDK will not use the commonly seen four parts versioning. For example a version sometimes refered to 10.19.0.0, will in the WebSDK be displayed as 1019.0.0. For simplicity this guide will only show three parts versioning numbers.
{{< /c8y-admon-info >}} 

We recommend to use the `^` or `~` in the `package.json` for all `@c8y` liberaries. When you are using a yealy long term support release, it is best to use the npm tag in your `package.json`. The long term support versions always end with `-lts`.

As our releases are bound to the Angular versioning, you must to ensure that you scaffold the right Angular version. Otherwise the scaffolding process will fail and give you a peer dependency error. The following table shows an overview of the supported versions:

| Angular version | Web SDK version | Comment |
| --- | --- | --- |

> | 16.x.x | 1019.x.x | |

Reviewer 2: Mention `ng cli` tooling?
